### PR TITLE
perl-www: add dependency on perl-try-tiny

### PR DESCRIPTION
**Pull request by tofurky**
Maintainer: me / @Naoir
Compile tested: x86_64, 19.07.5
Run tested: x86_64, 19.07.5

Description:

The perl-www bump from 6.15 -> 6.39 per https://github.com/openwrt/packages/commit/a9ad795 missed a new dependency on Try::Tiny (introduced in 6.17) which doesn't exist in packages.git. This adds a new package, perl-try-tiny, and changes the perl-www Makefile to depend on it.

### CODE REVIEW

File: lang/perl-try-tiny/Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(TOPDIR)/feeds/packages/lang/perl/perlmod.mk
+
+define Package/perl-try-tiny
+  SUBMENU:=Perl
+  SECTION:=lang
+  CATEGORY:=Languages
+  TITLE:=Minimal try/catch with proper preservation of $$$$@
```
Comment (neheb) [inline lang/perl-try-tiny/Makefile:30]: What
's $$$$@ ?

Comment (tofurky) [reply lang/perl-try-tiny/Makefile:30]: the actual perl variable is $@, but it needs to be escaped to show up properly in menuconfig. $ shows nothing, $$ also shows nothing, $$$$ did the trick :)

Comment (tofurky) [reply lang/perl-try-tiny/Makefile:30]: it was tough to properly preserve $@ ;)

**Comment by tofurky** [comment]
there was an issue with the DEPENDS for perl-try-tiny:
```
DEPENDS:=+perl +perlbase-essential
```

i changed it to this and force-pushed:
```
DEPENDS:=perl +perlbase-essential
```

**Comment by pprindeville** [comment]
@tofurky Seeing across the board build failures in CI/CD resulting from `perl-try-tiny`.  Please fix ASAP:

https://github.com/openwrt/packages/pull/14758/checks?check_run_id=1906631217

**Comment by tofurky** [comment]
> @tofurky Seeing across the board build failures in CI/CD resulting from `perl-try-tiny`. Please fix ASAP:
> 
> https://github.com/openwrt/packages/pull/14758/checks?check_run_id=1906631217

any idea why it's only showing up sometimes? i can't repro locally from current master. many other PRs have successful builds with the same thing, including this one 4 hrs ago:

image

if you could get me the contents of a the referenced logs/feeds/packages_ci/lang/perl-try-tiny/dump.txt, i could take a look at what's going on.

btw, the build you linked appears to be failing for a reason other than that spurious Makefile error...
```
2021-02-15T23:08:01.7981375Z find /home/build/openwrt/build_dir/target-x86_64_musl/libubox-2020-12-12-35787769/ipkg-x86_64/libblobmsg-json -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
2021-02-15T23:08:01.8390588Z Package libblobmsg-json is missing dependencies for the following libraries:
2021-02-15T23:08:01.8398541Z libubox.so
2021-02-15T23:08:01.8404374Z make[2]: *** [Makefile:119: /home/build/openwrt/bin/packages/x86_64/base/libblobmsg-json20201212_2020-12-12-35787769-1_x86_64.ipk] Error 1
2021-02-15T23:08:01.8405530Z make[2]: Leaving directory '/home/build/openwrt/feeds/base/package/libs/libubox'
2021-02-15T23:08:01.8413324Z time: package/feeds/base/libubox/compile#5.95#1.58#7.66
2021-02-15T23:08:01.8423565Z     ERROR: package/feeds/base/libubox failed to build.
2021-02-15T23:08:01.8429344Z make[1]: *** [package/Makefile:114: package/feeds/base/libubox/compile] Error 1
2021-02-15T23:08:01.8430352Z make[1]: Leaving directory '/home/build/openwrt'
2021-02-15T23:08:01.8437687Z make: *** [/home/build/openwrt/include/toplevel.mk:223: package/strongswan/compile] Error 2
```

**Comment by pprindeville** [comment]
Odd.  My `tmp/.packaginfo` says:

```
...
Package: libblobmsg-json
Version: 2020-12-12-357877693ca363b12e6e7e14d345639b2440cd07-1
Depends: +libc +USE_GLIBC:librt +USE_GLIBC:libpthread +libjson-c +libubox
Conflicts: 
Menu-Depends: 
Provides: 
Section: libs
Category: Libraries
Repository: base
Title: blobmsg <-> json conversion library
Maintainer: Felix Fietkau <nbd@nbd.name>
Source: libubox-2020-12-12-357877693ca363b12e6e7e14d345639b2440cd07.tar.xz
License: ISC
Type: ipkg
Description: blobmsg <-> json conversion library
Felix Fietkau <nbd@nbd.name>
@@
...
```

**Comment by tofurky** [comment]
https://github.com/openwrt/packages/pull/14761 hopefully fixes the intermittent makefile parsing issue.

File: lang/perl-try-tiny/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+PKG_SOURCE:=Try-Tiny-$(PKG_VERSION).tar.gz
+PKG_HASH:=da5bd0d5c903519bbf10bb9ba0cb7bcac0563882bcfe4503aee3fb143eddef6b
+PKG_BUILD_DIR:=$(BUILD_DIR)/perl/Try-Tiny-$(PKG_VERSION)
+
+PKG_MAINTAINER:=Matt Merhar <mattmerhar@protonmail.com>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include $(TOPDIR)/feeds/packages/lang/perl/perlmod.mk
```
Comment (pprindeville) [inline lang/perl-try-tiny/Makefile:24]: Is there a reason to not use `../perl/perlmod.mk`?  This is the convention everywhere else.

Comment (tofurky) [reply lang/perl-try-tiny/Makefile:24]: yes, building with an out of tree (local) packages repo. the PR i already opened changes it, though.

**Pull request merged**